agent/api/statechange.go

```diff
@@ -171,36 +171,30 @@ func newUncheckedContainerStateChangeEvent(task *apitask.Task, cont *apicontaine
 		ImageDigest:   cont.GetImageDigest(),
 		Reason:        reason,
 		Container:     cont,
-		Type:          eventType,
 	}
 	return event, nil
 }
 
-// NewManagedAgentChangeEvent creates a new special container state change event to convey managed agent state changes
+// NewManagedAgentChangeEvent creates a new managedAgent change event to convey managed agent state changes
 // returns error if the state change doesn't need to be sent to the ECS backend.
-func NewManagedAgentChangeEvent(task *apitask.Task, cont *apicontainer.Container) (ContainerStateChange, error) {
-	var changedManagedAgents []ManagedAgentStateChange
-	for _, ma := range cont.GetManagedAgents() {
-		if ma.Status.ShouldReportToBackend() {
-			changedManagedAgents = append(changedManagedAgents, ManagedAgentStateChange{
-				Name:   ma.Name,
-				Status: ma.Status,
-				Reason: ma.Reason,
-			})
-		}
+func NewManagedAgentChangeEvent(task *apitask.Task, cont *apicontainer.Container, managedAgentName string, reason string) (ManagedAgentStateChange, error) {
+	var event = ManagedAgentStateChange{}
+	managedAgent, ok := cont.GetManagedAgentByName(managedAgentName)
+	if !ok {
+		return event, errors.Errorf("No execute command agent available in container: %v", cont.Name)
```

Comment:
nit: we should change the log statement to print `managedAgentName` instead of hardcoding it as `execute command agent` (future proof)

Reply:
Agree -- I have a follow-up to fix the managed agent sent status.  will update there.